### PR TITLE
Do Not Review - Default article Template

### PR DESCRIPTION
**Pull request by killanaca**
- added fontFamily property to fonts collection inside the `fetch-fonts` task to fix an issue with bold and italic variants of the same font family
- removed `storybooknative-tvOS`, `storybooknative-tvOSTests`, `storybooknativeTests` and relative targets (and errors) from the Xcode project to deploy the storybooknative app on a real iphone device
- integration of the Default Article component

FIX on base component
**article-flag**
    use unicode space (to mimic the letter spacing just for android) and update lineHeigth, letterSpacing and font-size

**article-label**
    use unicode space (to mimic the letter spacing just for android) and update lineHeigth, letterSpacing and font-size

**article-byline**
    fix lineHeigth (13px for web, 17 for android, 14 for ios)

**date-publication**
    fix lineHeigth

### CODE REVIEW

File: packages/article/__tests__/article.test.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+/* eslint-env jest */
```
Comment (sericaia) [inline packages/article/__tests__/article.test.js]: Not 100% related to this PR but in the future we may want to add the following lines to .eslintrc.json
`
  "env": {
    "jest": true
  }
`

File: packages/article/__tests__/article.test.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import Article from "../article";
+
+const articleFixtureFull = require("../fixtures/full-article.json");
+const articleFixtureNoFlags = require("../fixtures/article-no-flags.json");
+const articleFixtureNoFlagsNoStandfirst = require("../fixtures/article-no-flags-no-standfirst.json");
+
+describe("Article test", () => {
+  it("renders native correctly", () => {
+    jest.mock("WebView", () => "WebView");
+    ReactNative.Platform.OS = "ios";
```
Comment (sericaia) [inline packages/article/__tests__/article.test.js]: maybe you could add the lines 14-15 to a `beforeEach`, all tests use it!

btw, how about android?

File: packages/article/__tests__/article.test.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+    const tree = renderer.create(<Article />).toJSON();
+    expect(tree).toMatchSnapshot();
+
+    jest.resetModules();
```
Comment (sericaia) [inline packages/article/__tests__/article.test.js]: add to afterEach :)

File: packages/article/article-style.js [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+  container: {
+    alignItems: "center",
+    flex: 1,
+    flexBasis: "100%",
+    justifyContent: "center"
+  }
+});
+
+const webStyles = StyleSheet.create({
+  // ArticleAd: {
```
Comment (sericaia) [inline packages/article/article-style.js]: remove these comments?

File: packages/article/article-style.js [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+  //   borderBottomWidth: 1,
+  //   paddingBottom: 15,
+  //   paddingTop: 15,
+  //   borderBottomColor: "#DBDBDB"
+  // },
+  ArticleMainContentRow: {
```
Comment (sericaia) [inline packages/article/article-style.js]: not sure if we should call this `ArticleMainContentRow` or `articleMainContentRow` for consistency. Can you ask @craigbilner ?

File: packages/article/article-style.js [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+  },
+  CaptionWrapper: {
+    paddingLeft: 10,
+    paddingRight: 10
+  },
+  ArticleTextElement: {
+    fontFamily: "TimesDigitalW04",
+    lineHeight: 1.53,
+    fontSize: 17,
+    // reset the default p style just for the web (on the native we have <Text> element with no default style)
```
Comment (sericaia) [inline packages/article/article-style.js]: remove comment :)

File: packages/article/article-style.js [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+    fontStyle: "normal",
+    lineHeight: 20,
+    fontSize: 18,
+    color: "#333"
+  }
+});
+
+const styles = {};
+
+if (Platform.OS === "android" || Platform.OS === "ios") {
```
Comment (sericaia) [inline packages/article/article-style.js]: maybe you can have only one `if` condition using Platform.OS === "web" ?

File: packages/article/article.js [hunk 8]
```diff
@@ -1,10 +1,234 @@
...
+import {
+  Text,
+  View,
+  ListView,
+  Platform,
+  StyleSheet,
+  ActivityIndicator
+} from "react-native";
+import PropTypes from "prop-types";
+import Ad, { AdComposer } from "@times-components/ad"; // , { AdComposer }
```
Comment (sericaia) [inline packages/article/article.js]: Remove comment. I would like to test all this locally before merging this PR ;)

File: packages/article/article.js [hunk 9]
```diff
@@ -1,10 +1,234 @@
...
+import get from "lodash.get";
 
-const Article = () =>
-  <View>
-    <ArticleContent code="intervention" />
-  </View>;
+import styles from "./article-style";
 
-export default Article;
+// const multiParagraph = require("./fixtures/multi-para.json").fixture;
```
Comment (sericaia) [inline packages/article/article.js]: old comment?

File: packages/article/article.js [hunk 10]
```diff
@@ -1,10 +1,234 @@
...
+              {React.cloneElement(el, {
+                style: StyleSheet.flatten([styles.ArticleTextElement])
+              })}
+            </View>
+          )}
+        </View>
+      );
+    }
+
+    return null;
```
Comment (sericaia) [inline packages/article/article.js]: quick note: this component is getting huge. Maybe we can split it into header, body, etc ?

File: packages/article/article.stories.js [hunk 11]
```diff
@@ -1,30 +1,60 @@
...
-const renderIframe = () => {
-  if (Platform.OS !== "web") {
-    return null;
+networkInterface.use([
+  {
+    applyMiddleware(req, next) {
+      if (!req.options.headers) {
+        req.options.headers = {};
+      }
+      req.options.headers["X-authorization"] = "DO_NOT_USE_IN_PROD";
```
Comment (sericaia) [inline packages/article/article.stories.js]: I think this will no longer work with the new ACS merge https://github.com/newsuk/times-public-api/pull/85

Comment (sericaia) [reply packages/article/article.stories.js]: @craigbilner  ?

File: packages/article/package.json [hunk 12]
```diff
@@ -52,15 +52,31 @@
...
+    "react-dom": "15.4.2",
+    "react-native": "0.42.3",
     "react-native-web": "0.0.119"
   },
   "peerDependencies": {
+    "@storybook/react-native": ">=3",
     "react": ">=15",
     "react-dom": ">=15",
-    "react-native": ">=0.42"
+    "react-native": ">=0.42",
```
Comment (sericaia) [inline packages/article/package.json]: please fix the version here :)

File: packages/provider/article-provider.js [hunk 13]
```diff
@@ -0,0 +1,34 @@
+import gql from "graphql-tag";
```
Comment (sericaia) [inline packages/provider/article-provider.js]: is this file a duplication of packages/article/article-provider.js ?

Comment (killanaca) [reply packages/provider/article-provider.js]: Fixed

**Comment by sericaia** [comment]
author-profile, card and provider also need a separate PR https://github.com/newsuk/times-components/pull/194/files#diff-e50ada87844019caa5d026ee882eeb23

**Comment by sericaia** [comment]
Closed due to https://github.com/newsuk/times-components/pull/226

**Pull request closed without merging**